### DISCUSSION TITLE
Integrate with mobile and another computers by Valent/KDEConnect.

### DISCUSSION THREAD

**Discussion by kacperpaczos** [andyholmes/valent#505] [Feature Request]
I created FR in GOA: https://gitlab.gnome.org/GNOME/gnome-online-accounts/-/issues/257

And I want to initiate a discussion/ask: Is it possible to seamlessly integrate devices with Valent? Is it science fiction or something real?
![image](https://github.com/andyholmes/valent/assets/23406555/e05e2035-fcfd-4899-92dd-4f0235358d3e)

**Comment by andyholmes** [comment]
GNOME Online Accounts is unfortunately pretty orthogonal to KDE Connect. It also doesn't support out-of-tree plugins, the last time I checked, so would be difficult to integrate in the way I think you're suggesting.

GNOME Online Accounts is built around centralized data sources like Google and other cloud providers. KDE Connect is based on a peer-to-peer, pseudo-mesh network. It is also geared toward simpler versions of things like contact sync and remote input, and will likely never replace full solutions for these features like RDP/VNC provide. It just can't communicate the type of detailed information these services expect.

Sorry!

**Reply by andyholmes** [reply]
On the other hand, in case it's not obvious, Valent *uses* GNOME Online Accounts to get things like contacts and sync with other devices.

**Comment by kacperpaczos** [comment]
Ok, i understand, but all what i need is a GUI to connect with my phone or device using Valent in the same place where i can connect to other web or network services. Just user experience.

And the next step is to make valent which works as a daemon, and provides API to Nemo/Nautilus (another subsystems) so i can make extensions in Python for example, to right-click actions to send file to another device, or else actions in my system, present seamless from my DE.

**Reply by andyholmes** [reply]
> Ok, i understand, but all what i need is a GUI to connect with my phone or device using Valent in the same place where i can connect to other web or network services. Just user experience.

Well, to get right to the point: this is not possible. GNOME Online Accounts doesn't support out-of-tree plugins, and requires them to be built-in. Even if Valent were the type of thing to put there, it's still in pre-alpha.

> And the next step is to make valent which works as a daemon, and provides API to Nemo/Nautilus (another subsystems) so i can make extensions in Python for example, to right-click actions to send file to another device, or else actions in my system, present seamless from my DE.

Then good news, because this is already the case! Valent is a service exposed over D-Bus, and both Nemo and Nautilus have their own plugin systems that can be used to interact with Valent.
